library/core/functions.general.php

```diff
@@ -1536,7 +1536,7 @@ function getRecord($recordType, $id, $throw = false) {
                     $Discussion = $Model->getID($Row['DiscussionID']);
                     if ($Discussion) {
                         $Discussion->Url = DiscussionUrl($Discussion);
```

Comment:
Probably. But 2 lines below the discussion is added to the record, so it may be used:
```php
$Row['Discussion'] = (array)$Discussion;
```